### DISCUSSION TITLE
Functions with dashes in name not showing up in right sidebar

### DISCUSSION THREAD

**Discussion by lognaturel** [pradyunsg/furo#767] [Potential Issue] [answered]
I'm documenting functions with dashes in their names:

```
.. function:: count-non-empty(nodeset)

  Returns the number of non-empty members of ``nodeset``.
```

You can see [on the rendered site](https://docs.getodk.org/form-operators-functions/#count) that `count-non-empty` is not in the right sidebar and does not have an anchor associated with it.

I [posted on Stackoverflow](https://stackoverflow.com/questions/77924112/show-function-names-with-dashes-in-sidebar-with-sphinx-and-function-directive) and was told this could be a theme issue.

**Comment by lognaturel** [comment]
It occurred to me that I could quickly try a different theme. I've confirmed it's an issue there too.

**Reply by dgw** [reply, marked as answer]
If no language domain is selected, Sphinx uses Python by default. (https://www.sphinx-doc.org/en/master/usage/domains/index.html)

Python does not allow `-` in identifiers, including function names, so it could be tripping on that. I see [an old config file for Sphinx](https://github.com/getodk/docs/blob/de527759e1507c98f10bfba7102bddaebdd76b16/docs/conf.old.py#L118) that specifies the `js` domain instead with a comment about syntactic compatibility, but that's as far as I've been able to dig.

**Reply by lognaturel** [reply]
Thanks so much for those tips!

**Reply by lognaturel** [reply]
Much gratitude to you @dgw, that's exactly what that config was for and unfortunately it was omitted when migrating to Furo.

I'd love to give you some internet points if you're on Stackoverflow!

**Reply by dgw** [reply]
Awesome that my advice helped you solve the problem! I summarized this on the Stack Overflow question you linked in the OP, so it won't look so lonely :)
